plugins/woocommerce-admin/client/products/sections/product-details-section.tsx

```diff
@@ -74,33 +81,35 @@ export const ProductDetailsSection: React.FC = () => {
 							{ ...getTextControlProps(
 								getInputProps( 'name' )
 							) }
+							onBlur={ () => {
```

Comment:
Instead of tying this to the name field's blur, could we hook things up to set the SKU if empty when the underlying name data changes?

Reply:
Great idea!  I'll give this a go.

Reply:
Sweet! I really think that is the way to go, as it keeps the different sections more decoupled and is also how ideally extensions would hook into things.

Reply:
There's not a very straightforward to do this since watching `values` returns changes on each character addition.  I opted to watch for previous touched state and compare to the current touched state of "name."

While this does not give us a 1 to 1 for blur events, it performs reasonably similarly and should accomplish the criteria needed for this PR.

https://github.com/woocommerce/woocommerce/pull/34978/commits/d08c11b47a09e24a7f632d2f71ed4682b75d342c